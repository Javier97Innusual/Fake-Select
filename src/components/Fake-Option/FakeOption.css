.select-option {
    padding: 1rem;
    border-bottom: 2px solid black;
    font-weight: bold;
    cursor: pointer;

    &:last-child {
        border-bottom: none;
    }

    &.select-hidden {
        display: none;
    }
}