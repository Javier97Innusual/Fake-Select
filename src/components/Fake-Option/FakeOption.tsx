import { FakeOptionType } from "./FakeOption.type";
import './FakeOption.css';

export function FakeOption({ value, children, hidden }: FakeOptionType) {
    return (
        <div
            className="select-option"
            data-id={value}
        >
            {children}
        </div>
    )
}