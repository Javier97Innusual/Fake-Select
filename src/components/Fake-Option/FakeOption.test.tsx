import { render } from '@testing-library/react';
import { FakeOption } from './FakeOption';

describe('FakeOption component', () => {
    it('should render correctly and match snapshot when hidden is false', () => {
        const { asFragment } = render(
            <FakeOption value="1">Option 1</FakeOption>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly and match snapshot when hidden is true', () => {
        const { asFragment } = render(
            <FakeOption value="1" hidden>Option 1</FakeOption>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should apply the hidden class when hidden prop is true', () => {
        const { container } = render(
            <FakeOption value="1" hidden>Option 1</FakeOption>
        );

        expect(container.firstChild).toHaveClass('select-hidden');
    });

    it('should not apply the hidden class when hidden prop is false', () => {
        const { container } = render(
            <FakeOption value="1">Option 1</FakeOption>
        );

        expect(container.firstChild).not.toHaveClass('select-hidden');
    });
});