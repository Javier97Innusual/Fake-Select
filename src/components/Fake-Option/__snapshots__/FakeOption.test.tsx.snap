// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeOption component should render correctly and match snapshot when hidden is false 1`] = `
<DocumentFragment>
  <div
    class="select-option"
    data-id="1"
  >
    Option 1
  </div>
</DocumentFragment>
`;

exports[`FakeOption component should render correctly and match snapshot when hidden is true 1`] = `
<DocumentFragment>
  <div
    class="select-option select-hidden"
    data-id="1"
  >
    Option 1
  </div>
</DocumentFragment>
`;
